FilePersistence/src/version_control/Commit.cpp

```diff
@@ -103,4 +103,88 @@ bool Commit::getFileContent(QString fileName, const char*& content, int& content
 		return false;
 }
 
+QStringList Commit::regexSearchNodeData(QString id) const
+{
+	QStringList matches;
+	for (auto file : files())
+	{
+		int index = 0;
+		auto content = QString::fromUtf8(file->content());
+
+		if ((index = content.indexOf(id, index)) != -1)
+		{
+			int flag = 0;
+			auto start = index;
+			auto end = index;
+
+			// start is the first char of the line containing id
+			while (start >= 0 && content[start] != '\n')
+			{
+				//string is of the form {.*} {id}
+				if (content[start] == '}')
+				{
+					flag = 1;
+					break;
+				}
+				start--;
+			}
+			if (flag == 1)	continue;
+			start++;
+
+			// end is the character after the line containing id
+			while (end <= content.size() && content[end] != '\n')	end++;
+
+			QString match = file->relativePath_ + ":" + content.mid(start, end-start);
+			matches << match;
+
+			// Find the next match
+			index = index+1;
+		}
+	}
+
+	return matches;
+}
+
+QStringList Commit::regexSearchNodeChildrenData(QString id) const
```

Comment:
Same comments like the function above: use `NodeIdType` and avoid regex in the name